crates/valv/src/lib.rs

```diff
@@ -159,19 +194,66 @@ impl ValvAPI for Valv {
             ..Default::default()
         };
 
-        self.db
-            .append_key_version(tenant.as_str(), key.clone(), key_version)
-            .await?;
+        let trx_result = self
+            .db
+            .database
+            .run(|trx, _| {
+                let trx = trx;
+                let tenant = tenant.clone();
+                let key = key.clone();
+                let key_version = key_version.clone();
+
+                async move {
+                    self.db
+                        .update_key_metadata(&trx, tenant.as_str(), key.clone())
```

Comment:
We use it twice When not using references (&), ownership is given away. Hence clone.
This can definitely be optimized in the future to reduce copying, but it's going to be less ergonomic.